### PR TITLE
Fix Set Tenant & Mock Driver

### PR DESCRIPTION
**Pull request by ArlonAntonius**
There's this issue where if you set the tenant it doesn't actually switch the database because we didn't tell the ConnectionResolver / DatabaseResolver we wanted to switch it.
This PR does that and add a Mock Driver.

### CODE REVIEW

File: src/Database/Mock/composer.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        "psr-4": {
+            "Tenancy\\Database\\Drivers\\Mock\\": ""
+        }
+    },
+    "authors": [
+        {
+            "name": "Daniël Klabbers",
+            "email": "daniel@klabbers.email",
+            "homepage": "http://luceos.com"
+        }
```
Comment (luceos) [inline src/Database/Mock/composer.json]: Missing an author.

File: src/Tenancy/Concerns/ResolvesTenants.php [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 trait ResolvesTenants
 {
-    protected function resolver(): Resolver
```
Comment (luceos) [inline src/Tenancy/Concerns/ResolvesTenants.php:21]: Why exactly is this necessary?

**Review by luceos — CHANGES_REQUESTED** [review]
At a quick glance this looks okay, but I dislike having a separate package for the db mock. Please move that code into the testing package instead. Otherwise we need to create a new package and make it a dependency for the testing package, which imo doesn't make much sense.

Few other remarks are in review.

File: src/Tenancy/Concerns/ResolvesTenants.php (hunk 2 above)
Comment (ArlonAntonius) [reply src/Tenancy/Concerns/ResolvesTenants.php:21]: Just made more sense if you have 2 resolvers that you use in 1 class to name them appropriately.

File: src/Database/Mock/composer.json (hunk 1 above)
Comment (ArlonAntonius) [reply src/Database/Mock/composer.json]: Won't matter if we get rid of the Mock Driver / place it into the testing package anyways.

**Review by luceos — CHANGES_REQUESTED** [review]
This change will cause that unsetting the tenant will keep the connection of the previous tenant active.

**Review by luceos — APPROVED** [review]

File: src/Testing/Concerns/CreatesApplication.php [hunk 3]
```diff
@@ -19,6 +19,7 @@
 use Illuminate\Database\Eloquent\Factory;
 use Tenancy\Environment;
 use Tenancy\Providers\TenancyProvider;
+use Tenancy\Database\Drivers\Mock\Providers\DatabaseProvider;
```
Comment (ArlonAntonius) [inline src/Testing/Concerns/CreatesApplication.php]: This is not used now, should remove it?

Comment (luceos) [reply src/Testing/Concerns/CreatesApplication.php]: we always remove unused imports

**Pull request merged**